### DISCUSSION TITLE
Correct syntax for WS2812B and ESP32

### DISCUSSION THREAD

**Discussion by ghost** [Makuna/NeoPixelBus#572] [Q&A] [answered]
On my board I have some WS2812B connected to GPIO33 of en ESP32.
I'm trying to enable the hw method, so I did:

```
NeoPixelBus<NeoGrbFeature, NeoEsp32I2s1800KbpsMethod> strip(8);
strip.Begin(-1, -1, 33, -1);
strip.ClearTo(green);
strip.Show();
```

But it does not compile with the following error:

>no matching function for call to 'NeoEsp32I2sMethodBase<NeoEsp32I2sSpeed800Kbps, NeoEsp32I2sBusOne, NeoEsp32I2sNotInverted>::Initialize(int8_t&, int8_t&, int8_t&, int8_t&)'

The signature of the function seems correct to my parameters.
What is the correct feature/method syntax for my use case?

**Comment by Makuna** [comment, marked as answer]
(This was not an issue, issues are for bugs, not questions, it was moved to discussions)

The Wiki area for the API includes the [constructors](https://github.com/Makuna/NeoPixelBus/wiki/NeoPixelBus-object-API#constructors) and [Begin](https://github.com/Makuna/NeoPixelBus/wiki/NeoPixelBus-object-API#void-begin).  

The signatures are not correct at all.  WS2812b are not DotStars (two wire smart LEDS).  This library supports both Two Wire and One Wire.  NeoPixels likes WS2812 are one wire smart LED.  The features and methods are different  for the two, and the constructor and Begin that you call are also different.  If you review the wiki and examples you can see examples uses for both.

Using the more specific method name (the one you uses would work also).... 

```
NeoPixelBus<NeoGrbFeature, NeoEsp32I2s1Ws2812xMethod> strip(PixelCount, PixelPin);

void setup() {
    strip.Begin();
...
}
```
